homeassistant/components/ecobee/climate.py

```diff
@@ -468,6 +461,11 @@ def preset_mode(self):
                 continue
 
             if event["type"] == "hold":
+                if event["holdClimateRef"] == "away" and is_indefinite_hold(
+                    event["startDate"], event["endDate"]
+                ):
+                    return PRESET_AWAY_INDEFINITELY
```

Comment:
From your PR description it's a bit hard to understand if there is two `away` modes but you only think we should one or if there is actually only one and why should we then not just use the built-in `away`?

Reply:
It appears the issue is a result of being able to pass both "away" and "Away" to the ecobee integration prior to the preset validation and their handling as they were treated differently within the code.  If passed as "away", the following code block was executed:

```
if preset_mode == PRESET_AWAY:
    self.data.ecobee.set_climate_hold(
        self.thermostat_index, "away", "indefinite", self.hold_hours()
    )
```

This would result in an indefinite hold regardless of what a user has configured for their ecboee schedule within the ecobee app and will have to be cancelled explicitly.  

Now if we pass "Away", we no longer have a match with `PRESET_AWAY` and the following code block is executed:

```
else:
    self.data.ecobee.set_climate_hold(
        self.thermostat_index,
        preset_mode,
        self.hold_preference(),
        self.hold_hours(),
    )
```

The default hold duration for an ecobee device is set to "Until the next scheduled activity" aka `nextTransition` and unless changed by the end user via the ecobee app it appears `hold_preference()` will return `nextTransition` which would set preset mode back to whatever the schedule indicates when the time arrives.  

I can't make sense of the following code block as `PRESET_HOLD_NEXT_TRANSITION` and `PRESET_HOLD_INDEFINITE`  are not preset modes but hold types used to determine the duration of a hold although I honestly didn't delve into pyecobee to see if it handles it but with the recent validation I'm unsure if it would work regardless.  

```
elif preset_mode in (PRESET_HOLD_NEXT_TRANSITION, PRESET_HOLD_INDEFINITE):
            self.data.ecobee.set_climate_hold(
                self.thermostat_index,
                PRESET_TO_ECOBEE_HOLD[preset_mode],
                self.hold_preference(),
                self.hold_hours(),
            )
```

Once preset mode validation was implemented this unintended interaction between "away"/"Away" and their different handling within the integration(or pyecobee perhaps) broke things for people who utilized this interaction in their automations.  Seems like we could expose the hold type as a parameter for the `set_preset_mode()`, setting it to 'nextTransition' by default to follow ecobee defaults but allow it to be overridden with `indefinite` when a permanent hold is required.  

I hope this helps gjohansson-ST(and others) and if I made any mistakes in my statements regarding the code(hobby coder), please feel free to correct me for my betterment and thank you.

Reply:
@afxefx you're pretty much spot on - thanks for adding a more thorough explanation.

pyecobee just blindly sends what it's given to the ecobee API, so your intuition is correct - that code block makes no sense, would never have worked, and so I removed it as part of this change (in addition to the fact that it's now also entirely inaccessible due to the newly-added preset mode validation).

I considered expanding the ecobee HA service with an optional hold type parameter, but I think that creates more confusion than it's worth (by deviating from the base `Climate` component) and would create the need to handle a variety of preset + hold duration pairs, many of which don't make sense. Realistically, it seems the only thing that users want is an indefinite away hold, so this simpler approach felt better to me.

@gjohansson-ST please pay careful attention to the casing of the preset names ("away" vs "Away") in the PR description - I believe it makes sense. I can add "lowercase" and "uppercase" notations in the description if you think that would be helpful.